### PR TITLE
Fix fit Rabi

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
This PR fix Rabi fit and refactor Rabi utils.
Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: src/qibocal/protocols/characterization/rabi/utils.py [hunk 1]
```diff
@@ -195,3 +188,13 @@ def plot_probabilities(data, qubit, fit):
...
+
+
+def extract_rabi(data):
+    """
+    Extract Rabi fit info.
+    """
+    if "RabiAmplitude" in data.__class__.__name__:
+        return "amp", "Amplitude (dimensionless)", rabi_amplitude_function
+    elif "RabiLength" in data.__class__.__name__:
+        return "length", "Time (ns)", rabi_length_function
```
Comment (AleCandido) [inline src/qibocal/protocols/characterization/rabi/utils.py:200]: Exhaust the options: if you're failing all conditions, and there is no meaningful default, just `raise` an error.

(currently, you're implicitly returning `None`)

**Review by andrea-pasquale — APPROVED** [review]
Thanks @Edoardo-Pedicillo, LGTM.

File: src/qibocal/protocols/characterization/rabi/utils.py [hunk 2]
```diff
@@ -193,3 +179,14 @@ def plot_probabilities(data, qubit, fit):
...
+    raise RuntimeError("Data has to be a data structure of the Rabi routines.")
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/characterization/rabi/utils.py:192]: If you can could add a quick test for this function to increase coverage.

**Review by GabrielePalazzo — APPROVED** [review]

File: src/qibocal/protocols/characterization/rabi/utils.py [hunk 3]
```diff
@@ -193,3 +179,14 @@ def plot_probabilities(data, qubit, fit):
...
     return figures, fitting_report
+
+
+def extract_rabi(data):
+    """
+    Extract Rabi fit info.
+    """
+    if "RabiAmplitude" in data.__class__.__name__:
+        return "amp", "Amplitude [dimensionless]", rabi_amplitude_function
+    elif "RabiLength" in data.__class__.__name__:
```
Comment (AleCandido) [inline src/qibocal/protocols/characterization/rabi/utils.py:190]:
```suggestion
    if "RabiLength" in data.__class__.__name__:
```

**Review by AleCandido — APPROVED** [review]

**Pull request merged**